robin/model/multimodal_encoder/open_clip.py

```diff
@@ -1,116 +1,68 @@
 import torch
-import torch.nn.functional as F
-from urllib.request import urlopen
-from PIL import Image
-from open_clip import create_model_from_pretrained, get_tokenizer # works on open-clip-torch>=2.23.0, timm>=0.9.8
-import open_clip
 from torch import nn
-import os
-from .encoder_info import OPENCLIP_CONFIG_MAP, TIMM_ON_OPENCLIP
+
+from open_clip import create_model_from_pretrained
+
 
 class OpenCLIPVisionTower(nn.Module):
     def __init__(self, vision_tower, args, delay_load=False):
         super().__init__()
-
-        self.is_loaded = False
+        assert args.mm_vision_select_layer == -1, "open clip support output tokens of last layer only"
 
         self.vision_tower_name = vision_tower
-        # self.select_layer = args.mm_vision_select_layer
-        # self.select_feature = getattr(args, 'mm_vision_select_feature', 'patch')
-
+        self.select_layer = args.mm_vision_select_layer
+        self.select_feature = getattr(args, 'mm_vision_select_feature', 'patch')
+        self.hidden_size = None # Place Holder get this value in load_model function
+        self.dtype = None # Place Holder get this value in train function
+        self.is_loaded = False
         if not delay_load:
             self.load_model()
-        else:
-            pass
-        
-        # self.hidden_size = 1152 if "SO400M" in self.vision_tower_name else 768  # TODO: this is a hack, we should read this from config.embed_dim
-
-        self.device = None
-        self.dtype = None
         
     def load_model(self):
-        name = self.vision_tower_name.split("/")[-1]
-        
-        if os.path.exists(self.vision_tower_name):
-
-            config = OPENCLIP_CONFIG_MAP[name] if name in OPENCLIP_CONFIG_MAP.keys() else name
-
-            self.vision_tower, self.image_processor = create_model_from_pretrained(config, pretrained=self.vision_tower_name+'/open_clip_pytorch_model.bin')
-        else:
-            self.vision_tower, self.image_processor = create_model_from_pretrained(self.vision_tower_name)
-        
-
+        assert "/" in self.vision_tower_name, """
+            model name and pretrained must be split with / in open clip eg: ViT-B-16/laion2b_s34b_b88k"""
         
-        self.vision_tower = self.vision_tower.visual
-
-        if "timm" in self.vision_tower_name or name.lower() in TIMM_ON_OPENCLIP:
-            self.hidden_size = self.vision_tower.trunk.embed_dim
```

Comment:
add test for `ViT-SO400M-14-SigLIP-384`

Reply:
Hi, i test at here, but one wired thing I found is that forward_features  return without cls token, is this because that siglip don't have such cls token, or something else?

 https://github.com/AGI-Collective/Robin/commit/431ac9c1bb7dc5d41731dfeacaf087d66e5c8dca